### PR TITLE
feat(expressive-modal/lightbox-media-viewer): update expressive modal v2

### LINKED ISSUE #10850: [Component] Expressive modal

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

Closes #10850, #10855

### Description

Updates `expressive-modal` and `lightbox-media-viewer` components to v2 styles and prefixes

### Changelog

**New**

- {{new thing}}

**Changed**

- update styles
- update component prefixes

**Removed**

- {{removed thing}}

### CODE REVIEW

**Review by ariellalgilmore — COMMENTED** [review]
Looking good! just a small change, but might also need to update the codesandbox examples, readme's, and react wrapper stories.

Oh also the expressive modal still stays "open" when closing
Screenshot 2023-10-19 at 1 22 39 PM

**Comment by kennylam** [comment]
@ariellalgilmore The `open` knob for the React wrapper stories is bugged for both components. Looks like this is a Storybook issue, since the `open` prop is clearly working when the story first loads. Also, the `hideCaption` prop isn't working in `lightbox-media-viewer`. This issue also exists in production. I'll create issues for both of these bugs separately.

File: packages/styles/scss/components/expressive-modal/_expressive-modal.scss [hunk 1]
```diff
@@ -15,22 +15,22 @@
...
 @mixin expressive-modal {
-  $modal: '.#{$prefix}--modal';
-  $button-size: $spacing-09;
   $icon-size: rem(20px);
-  $offset-close-icon: $spacing-07 - list.slash($button-size - $icon-size, 2);
 
-  :host(#{$c4d-prefix}-expressive-modal),
-  #{$modal}--expressive {
-    color: $text-primary;
+  // :host(#{$c4d-prefix}-expressive-modal:not([open='false']))
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/expressive-modal/_expressive-modal.scss:27]:
```suggestion
```

File: packages/web-components/src/components/expressive-modal/__stories__/expressive-modal.stories.ts [hunk 2]
```diff
@@ -60,9 +58,9 @@ export const Default = (args) => {
         sapien eu nisl congue suscipit.
       </c4d-expressive-modal-body>
       <c4d-expressive-modal-footer>
-        <c4d-button-expressive href="https://www.example.com">
+        <c4d-button is-expressive href="https://www.example.com">
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/expressive-modal/__stories__/expressive-modal.stories.ts:61]:
```suggestion
        <c4d-button href="https://www.example.com">
```

**Review by ariellalgilmore — APPROVED** [review]
LGTM!!

**Pull request merged**